indexer/etl/etl.go

```diff
@@ -146,3 +147,21 @@ func (etl *ETL) processBatch(headers []types.Header) error {
 	etl.etlBatches <- ETLBatch{Logger: batchLog, Headers: headersRef, HeaderMap: headerMap, Logs: logs.Logs, HeadersWithLog: headersWithLog}
 	return nil
 }
+
+// Determine the starting height for header traversal
+func getStartingBlock(header *database.BlockHeader, headerAtStart *types.Header) *types.Header {
+	var fromHeader *types.Header
+	startHeight := headerAtStart.Number
+
+	if header != nil {
+		log.Info("detected last indexed block", "number", header.Number, "hash", header.Hash)
```

Comment:
lets avoid usage of the global logger and pass in the log. It can even have the right `"etl", "(l1|l2)"` set on it by the respective etl